src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBinderSourceGenerator.Emitter.cs

```diff
@@ -0,0 +1,469 @@
+            {
+                if (_bindMethodGen_ProcessedTypes.Contains(type))
+                {
+                    return;
+                }
+
+                _bindMethodGen_ProcessedTypes.Add(type);
+
+                _writer.WriteBlockStart(@$"{visibility} static void {LiteralFor_Bind}(this {GlobalNameForIConfigurationType} {LiteralFor_configuration}, {type.ToDisplayString()} {LiteralFor_obj})");
+
+                if (type is IArrayTypeSymbol { } arrayType)
+                {
+                    EmitBindMethodImplForArray((arrayType.ElementType as INamedTypeSymbol)!);
+                }
+                else if (TypesAreEqual(type, _typeSymbolForIConfigurationSection))
+                {
+                    EmitCastToIConfigurationSection();
+                    EmitAssignment(LiteralFor_obj, LiteralFor_section);
+                }
+                else if (type is INamedTypeSymbol namedType)
+                {
+                    if (!type.IsValueType)
+                    {
+                        EmitCheckForNullArgument();
+                    }
+
+                    if (!IsCollection(namedType))
+                    {
+                        EmitBindMethodImplForObject(namedType);
+                    }
+                    else
+                    {
+                        CollectionBindingSpec? spec = GetCollectionBindingSpec(namedType);
+                        if (!spec.HasValue)
+                        {
+                            Emit_NotSupportedException_UnableToBindType(type, NotSupportedReason_CollectionNotSupported);
+                        }
+                        else if (spec.Value.KeyType is not null)
+                        {
+                            EmitBindMethodImplForDictionary(spec.Value);
+                        }
+                        else
+                        {
+                            EmitBindMethodImplForEnumerable(spec.Value);
+                        }
+                    }
+
+                    bool IsCollection(INamedTypeSymbol type)
+                    {
+                        return type.AllInterfaces.Contains(_typeSymbolForIEnumerable, SymbolEqualityComparer.Default) ||
+                            TypesAreEqual(type, _typeSymbolForIEnumerable);
+                    }
+                }
+
+                _writer.WriteBlockEnd();
+            }
+
+            private static void EmitBindMethodImplForArray(INamedTypeSymbol elementType)
+            {
+                Debug.Assert(elementType is not null);
+            }
+
+            private void EmitBindMethodImplForObject(INamedTypeSymbol type)
+            {
+                foreach (ISymbol member in type.GetMembers())
+                {
+                    if (member is IPropertySymbol
+                        {
+                            SetMethod: IMethodSymbol
+                            {
+                                DeclaredAccessibility: Accessibility.Public,
```

Comment:
>Public [](http://example.com/codeflow?start=69&length=6)

do we support `BinderOptions.BindNonPublicProperties = true`?

Reply:
Not yet, our current feeling is that we won't pick up the overloads that take `BinderOptions`.